### PR TITLE
dell-kestrel: introduce new plugin to support composite usb devices

### PR DESCRIPTION
**Pull request by CragW**
(cherry picked from commit 80019ff7aa53dfcaa05270db2428c159a2ce0800)

Type of pull request:

- [x] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/dell-kestrel/fu-dell-kestrel-ec.c [hunk 1]
```diff
@@ -0,0 +1,1079 @@
...
+			guint8 location;
+			guint8 device_type;
+			guint8 sub_type;
+			guint8 arg;
+			guint8 instance;
+		} ec_addr_map;
+		union {
+			guint32 version_32;
+			guint8 version_8[4];
+		} __attribute__((packed)) version; /* nocheck:blocked */
```
Comment (superm1) [inline plugins/dell-kestrel/fu-dell-kestrel-ec.c:60]: I guess the backport for the rs stuff was tough?  I'm fine with this knowing we have it done right in main, but I think @hughsie can make the call here.

Comment (CragW) [reply plugins/dell-kestrel/fu-dell-kestrel-ec.c:60]: > I guess the backport for the rs stuff was tough? 

I tried to pull the rsgen from main but several differencies are seen in the history, likely the backport are selected to apply. 

For broader stability on a stable branch I keep the changes in plugin alone as is.  See @hughsie if he plan to backport the new work in main.

Comment (superm1) [reply plugins/dell-kestrel/fu-dell-kestrel-ec.c:60]: I had a quick glance and it looks like a bunch of gymnastics to make it happen, maybe better for you to.

Comment (CragW) [reply plugins/dell-kestrel/fu-dell-kestrel-ec.c:60]: How about let's merge this PR and tag 1.9.27 to address the schedule and stability concerns?  Another tag with sufficient time for the rs work and adoption/testing.

Comment (hughsie) [reply plugins/dell-kestrel/fu-dell-kestrel-ec.c:60]: If we do that the follow-on work never gets done.

**Comment by superm1** [comment]
Can you please redo now that the rs stuff is backported?

**Comment by CragW** [comment]
> Can you please redo now that the rs stuff is backported?

I plan to close this PR, and create another to include the rs stuff.  In seconds.

**Pull request closed without merging**